pages/users-page.tsx

```diff
@@ -13,36 +13,36 @@ import { Section } from '../section';
 import { Styles } from '../styles';
 import { usersQueryGQL } from '../graphql/querys';
 import { client } from '../services/apollo';
-import { useQuery } from '@apollo/client';
+import { useLazyQuery } from '@apollo/client';
 
 export const UsersPage = () => {
   const isDarkMode = useColorScheme() === 'dark';
   const backgroundStyle = {
     backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
   };
 
-  function UsersQuery() {
-    const { loading, error, data } = useQuery(usersQueryGQL, { client: client });
-  
-    if (loading) {
-      return <ActivityIndicator/>;
-    }
-    if (error) {
-      return <Text style={Styles.errorMessage}>`Erro! ${error.message}`</Text>;
+  const [limit, setLimit] = useState(25);
+
+  const [usersQuery, {data, loading, error}] = useLazyQuery(usersQueryGQL,
+    {
+      client: client, 
+      onCompleted: () => {
+        setLimit(limit + 25);
+      },
+      onError: () => {
+        console.log(error);
+      }
     }
-    return (
-      <FlatList 
-        data={data.users.nodes}
-        renderItem={({item}) => (
-            <Text style={Styles.userList}>
-              Usuário: {item.name}{'\n'}
-              Email: {item.email}
-            </Text>
-        )}
-      />
-    );
+  );
```

Comment:
aqui é meio perigoso, vc ta puxando tudo novamente e mais 25 entradas, dependendo da api tem limite de quanto pode puxar de uma vez e ai essa solução não funciona. O apollo tem uma [documentação sobre paginação](https://www.apollographql.com/docs/react/pagination/overview), da uma olhada lá que tem bastante coisa interessante

Reply:
Como a intenção foi implementar o infinite scroll, acabei tendo essa dúvida em relação a qual volume de dados devo guardar, dado que se eu seguir a navegação padrão (rolagem para baixo) e for atualizando o offset conforme carrego novos itens, acabaria se perdendo os dados impossibilitando que o usuário rolasse a tela de volta pra cima, não?
Para este caso citado, não achei uma forma de "carregar novamente" a página caso o usuário rolasse para cima